packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue

```diff
@@ -1,18 +1,281 @@
+		return [
+			{
+				label: 'Select outputs to display in operator',
+				items: props.node.outputs
+			}
+		];
+	}
+	return [];
+});
+const selectedOutputId = ref<string>();
+const selectedConfigId = computed(
+	() => props.node.outputs?.find((o) => o.id === selectedOutputId.value)?.value?.[0]
+);
+
+const configCache = ref<Record<string, ModelConfiguration>>({});
+
+const activeIndex = ref<number>(0);
+const configName = ref<string>(props.node.state.configName);
+const configDescription = ref<string>(props.node.state.configDescription);
+const model = ref<Model>();
+
+const configInitials = ref<Initial[]>();
+const configParams = ref<ModelParameter[]>();
+
+onMounted(async () => {
+	const input = props.node.inputs[0];
+	if (input.value) {
+		const m = await getModel(input.value[0]);
+		if (m) {
+			model.value = m;
+
+			// TODO: set this only if configs don't exist already
+			// set the state with the model initials and params
+			const state = _.cloneDeep(props.node.state);
+			state.configInitials = m.semantics?.ode.initials;
+			state.configParams = m.semantics?.ode.parameters;
+			emit('update-state', state);
+		}
+	}
+});
+
+const updateState = (updatedField) => {
+	let state = _.cloneDeep(props.node.state);
+	state = { ...state, ...updatedField };
+	emit('update-state', state);
+};
+
+const updateConfigParam = (param) => {
+	const state = _.cloneDeep(props.node.state);
+	// find param with the same id and update it
+	state.configParams = state.configParams?.map((p) => {
+		if (p.id === param.id) {
+			return param;
+		}
+		return p;
+	});
+	emit('update-state', state);
+};
+
+const updateConfigInitial = (initial) => {
+	const state = _.cloneDeep(props.node.state);
+	// find initial with the same target and update it
+	state.configInitials = state.configInitials?.map((i) => {
+		if (i.target === initial.target) {
+			return initial;
+		}
+		return i;
+	});
+	emit('update-state', state);
+};
+
+const saveConfiguration = async () => {
+	if (model.value) {
+		const newModel = _.cloneDeep(model.value);
+		if (newModel.semantics) {
+			newModel.semantics.ode.initials = configInitials.value;
+			newModel.semantics.ode.parameters = configParams.value;
+		}
+
+		const data = await createModelConfiguration(
+			model.value.id,
+			configName.value,
+			configDescription.value,
+			newModel
+		);
+
+		setTimeout(async () => {
+			const modelConfig = await getModelConfigurationById(data.id);
+			configCache.value[modelConfig.id] = modelConfig;
+
+			emit('append-output-port', {
+				type: ModelConfigOperation.outputs[0].type,
+				label: modelConfig.name,
+				value: modelConfig.id,
+				isSelected: false,
+				state: {
+					modelId: modelConfig.modelId,
+					configName: modelConfig.name,
+					configDescription: modelConfig.description,
+					configInitials: modelConfig.configuration.semantics?.ode.initials,
+					configParams: modelConfig.configuration.semantics?.ode.parameters
+				}
+			});
+		}, 1000);
```

Comment:
This setTimeout mechanism is not really necessary, since you already have the values you want to propagate forward. 

Also it would be better named `createModelConfiguration` to better reflect what you are actually doing. Note there is another savings part; in the semantics-table that doesn't seem to work that we should sync up on. 

Nitpick: I might suggest inverting the logic, instead of
```
if (xyz) {
   ...do stuff...
}
```
This makes the code a little bit less steep in terms of indentation
```
if (!xyz) return
...do stuff...
```

Reply:
moved the `emit(append-output-port...` out of the `setTimeout` but keeping the `setTimeout` here for caching for now.